sdk/program/src/sysvar/slot_hashes.rs

```diff
@@ -125,22 +109,46 @@ impl SlotHashesSysvar {
             .map(u64::from_le_bytes)
             .ok_or(ProgramError::InvalidAccountData)?;
 
-        // If the vector length is 0, return an empty vector.
-        if slot_hash_count == 0 {
-            return Ok(vec![]);
-        }
-
         // From the vector length, determine the expected length of the data.
         let length = (slot_hash_count as usize)
             .checked_mul(std::mem::size_of::<PodSlotHash>())
             .ok_or(ProgramError::ArithmeticOverflow)?;
-        let start = U64_SIZE;
-        let end = start.saturating_add(length);
+        let slot_hashes_start = U64_SIZE;
+        let slot_hashes_end = slot_hashes_start.saturating_add(length);
 
-        // Finally, convert to a vector of `PodSlotHash`.
-        data.get(start..end)
+        Ok(Self {
+            data,
+            slot_hashes_start,
+            slot_hashes_end,
+        })
+    }
+
+    /// Get a value from the sysvar entries by its key.
+    /// Returns `None` if the key is not found.
+    pub fn get(&self, slot: &Slot) -> Result<Option<Hash>, ProgramError> {
+        self.get_pod_slot_hashes().map(|pod_hashes| {
+            pod_hashes
+                .binary_search_by(|PodSlotHash { slot: this, .. }| slot.cmp(this))
+                .map(|idx| pod_hashes[idx].hash)
+                .ok()
+        })
+    }
+
+    /// Get the position of an entry in the sysvar by its key.
+    /// Returns `None` if the key is not found.
+    pub fn position(&self, slot: &Slot) -> Result<Option<usize>, ProgramError> {
+        self.get_pod_slot_hashes().map(|pod_hashes| {
+            pod_hashes
+                .binary_search_by(|PodSlotHash { slot: this, .. }| slot.cmp(this))
+                .ok()
+        })
+    }
```

Comment:
These, along with the addition of fields to the struct, are breaking changes. Do you plan to break and backport the breaking change? We're still doing a little bit of that, and since it's a new API, it should be OK if that's how you want to go.

If that's the case, how about making `SlotHashesSysvar` implement the `Sysvar` trait for `get()`, and then add the other functions as before?

Or, we could avoid a breaking change by creating a new `PodSlotHashes` struct and deprecate the old one, which would make a nice analogue to the normal `SlotHashes`. What do you think?

Reply:
> Or, we could avoid a breaking change by creating a new `PodSlotHashes` struct and deprecate the old one, which would make a nice analogue to the normal `SlotHashes`. What do you think?

I think this is probably the better way to go. We can deprecate `SlotHashesSysvar` and add a `PodSlotHashes` that implements `Sysvar` and offers a similar API to `SlotHashes`. Nice idea!

I think it maybe makes more sense at this point to open a new PR where I can add the deprecation and new struct/API. Work for you?

Reply:
No problem, whatever you prefer!